sympy/printing/pycode.py

```diff
@@ -296,6 +296,14 @@ def _print_Sum(self, expr):
     def _print_ImaginaryUnit(self, expr):
         return '1j'
 
+    def _print_KroneckerDelta(self, expr):
+        a, b = expr.args
+
+        return '1 if ({a}) == ({b}) else 0'.format(
```

Comment:
I actually meant the entire expression should be in parentheses. For example, `pycode(KroneckerDelta(x, y) + 1)` is wrong. For the inner expression it should use `self.parenthesize`, although I think there won't actually be any cases where it is needed because `==` has very high precedence in Python.

Reply:
Ah, I see. I'm not sure I know how to use `self.parenthesize`… Should it be something like this?
```python
a, b = expr.args
PREC = precedence(expr)

a = self.parenthesize(a, PREC)
b = self.parenthesize(b, PREC)

return '(1 if {a} == {b} else 0)'.format(...)
```

Reply:
Yes, except I'm not sure `precedence(expr)` will work without a change to `precedence`. 

You can check with some operators that have higher precedence than `==` https://docs.python.org/3/reference/expressions.html#operator-precedence. The only relevant one I see that would be relevant are logic operations, which don't actually make sense inside of a KroneckerDelta, but I guess you can put them there to test.

Reply:
After reading through `precedence` and `parenthesize`, I think this code would've worked:
```python
a, b = expr.args
PREC = PRECEDENCE['Relational']

a = self.parenthesize(a, PREC)
b = self.parenthesize(b, PREC)

return '(1 if {a} == {b} else 0)'.format(...)
```
But the actual implementation of `KroneckerDelta` relies on the argument being numbers, not logical expressions, and it threw an error for `KroneckerDelta(And(...), ...)`. So I'm just leaving the `parenthesize` out and going with a simpler version.